oak_functions_service/src/lib.rs

```diff
@@ -86,6 +95,22 @@ impl OakFunctionsService {
             )
         })
     }
+    /// Extract the public key without verifying. This function is intended as a temporary
+    /// fix to allow the application to start using the oak SDK, while it still
+    /// requires the ability to extract the attestation public key, and to send it
+    /// directly to the client.
+    // TODO(#4571): Remove this function once the client has been updated to use
+    // the verification library. At this point the evidence will just be directly
+    // sent to the client, which extracts the public key from it upon succesful
+    // verification.
+    fn get_encryption_public_key(&self) -> Result<Vec<u8>, alloc::string::String> {
+        let encryption_claims = self.evidencer.get_evidence().application_keys.claims()?;
+        let encryption_cose_key =
+            oak_dice::cert::get_public_key_from_claims_set(&encryption_claims)
+                .map_err(|msg| msg.to_string())?;
+        oak_dice::cert::cose_key_to_hpke_public_key(&encryption_cose_key)
+            .map_err(|msg| msg.to_string())
+    }
```

Comment:
@ipetr0v should we move a version of this into the SDK?

Reply:
Is public key being sent to the client separately from the evidence?

If so, then yes. But such things are often hard to remove from the SDK afterwards, since everyone will start using it. So maybe clients should just extract the public key just from the evidence (I think we still send an old non-DICE version of the evidence that contains the public key)

Reply:
yes I think eventually we should just send the evidence to the client, and the client extracts the public key. 

We're not they're yet, and the client impl still expects to be passed a seperate client key. Hence this func exists to extract it in the app until the client is updated. 

Until the client is updated or until we have the orchestrator, all enclave apps for restricted kernel will need impl this.

Reply:
I think it's fine to leave out of the SDK though, since need for enclave apps to do this themselves will go away.